### PR TITLE
A25: Line completed

### PR DESCRIPTION
**Pull request by rajeevramanTMM**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -24,10 +24,59 @@ class MyWindow : Window {
...
+      if (swap) (x0, y0, x1, y1) = (y0, x0, y1, x1);
+      if (x0 > x1) (x0, y0, x1, y1) = (x1, y1, x0, y0);
+      dx = x1 - x0; dy = y1 - y0;
+      bool down = y0 > y1;
+      if (down) dy = -dy;
+      int diff = (2 * dy) - dx;
+      int y = y0;
+      for (int x = x0; x <= x1; x++) {
+         try {
+            mBmp.Lock ();
```
Comment (tarydon) [inline Program.cs]: Don't lock and unlock for every pixel - do a Lock / Unlock around the complete loop and you will see a big performance improvement.

File: Program.cs [hunk 2]
```diff
@@ -24,10 +24,59 @@ class MyWindow : Window {
...
+            mBase = mBmp.BackBuffer;
+            if (swap) {
+               SetPixel (y, x, 255);
+               mBmp.AddDirtyRect (new Int32Rect (y, x, 1, 1));
```
Comment (tarydon) [inline Program.cs]: Try to do a single AddDirtyRect for the entire line, instead of one per pixel.

**Review by tarydon — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 1 above)
Comment (rajeevramanTMM) [reply Program.cs]: Correction made

File: Program.cs (hunk 2 above)
Comment (rajeevramanTMM) [reply Program.cs]: Correction made

**Pull request closed without merging**